src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIPacket.cs

```diff
@@ -46,46 +36,58 @@ public string Description
         }
 
         /// <summary>
-        /// Data left to process
+        /// Length of data left to process
         /// </summary>
-        public int DataLeft
+        public int LengthOfDataLeft
         {
             get
             {
-                return _length - _offset;
+                return _dataLength - _offset;
             }
         }
 
         /// <summary>
-        /// Length of data
+        /// Length of buffer
         /// </summary>
-        public int Length
+        public int BufferLength
         {
             get
             {
-                return _length;
+                return _dataBuffer.Length;
             }
         }
 
         /// <summary>
-        /// Packet validity
+        /// Length of data in buffer
         /// </summary>
-        public bool IsInvalid
+        public int DatatLength
         {
             get
             {
-                return _data == null;
+                return _dataLength;
+            }
+        }
+
+        public bool IsEmpty
```

Comment:
`IsInvalid` is a better name that `IsEmpty` the reason being, a packet with a null buffer is considered Invalid. However a buffer with a non-null buffer but with buffer with no data will be considered empty.